docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,175 @@
+ - ONNX Runtime. ONNX Runtime doesn't have prebuilt packages for more than just x86/x64. It does support Arm, but we have to build it ourselves or get the ONNX Runtime team to package Arm assemblies. This is the same situation as with LightGBM.
+
+
+## 3 Possible Solutions
+We have several possible options we can use to resolve this:
+ - Keep ML.NET native components and rewrite them to avoid problematic dependencies completely.
+ - Keep ML.NET native components and ifdef/rewrite to avoid problematic dependencies on platforms/architectures only where they are not supported.
+ - Hybrid approach of replacement code and software fallback. This is the approach I recommend.
+
+None of these approaches resolve the 3rd party dependency issues. These solutions only deal with first party ML.NET code itself.
+
+I have lots more info about our dependency on x86/x64 in another document if required.
+
+### 3.1 Rewrite native components to work on other platforms
+This will still allow us to gain the benefits of the X86/x64 SIMD instructions and Intel MKL on architectures that support it but will also keep the benefits of native code in the other places. The downside is that we would have to build the native code for, potentially, a lot of different architectures.
+
+At a high level, this solution would require us to:
+ - Fix the build so it's not hardcoded to look for specific native dependencies.
+ - Fix the managed code so that if it can't find the native components the code behaves correctly.
+ - All the native code would need to have the x86/x64 dependencies change to other libraries or be re-written by us.
+   - CpuMathNative would need to be re-written to not use x86/x64 dependencies.
+   - FastTreeNative would need to be re-written to not use x86/x64 dependencies.
+   - LdaNative builds just find without x86/x64 dependencies, so no replacements would need to be found.
+   - MatrixFactorizationNative builds with the "USESSE" flag that requires x86/x64. We can conditionally enable/disable that flag and no other work would be required.
+   - MklProxyNative is only a wrapper for Intel MKL and can be ignored on non x86/x64 platforms. We will need to modify the build to exclude it as needed.
+   - SymSgdNative would need to be re-written to not use Intel MKL. There are only 4 methods that would need to be changed.
+
+I was unable to find all the replacements we would need. We would end up having to write many native methods ourselves with this approach.
+
+### 3.2 Rewrite native components to be only managed code
+This will truly allow ML.NET to run anywhere that .NET runs. The only downside is the speed of execution, and the time to rewrite the existing native code we have.  If we restrict new architectures to .NET core 3.1 or newer, we will have an easier time with the software fallbacks as some of this code has already been written. This solution will also require a lot of code rewrite from native code to managed code.
```

Comment:
I believe with .NET Core 3.1 you have issues with WebAssembly because it looks for CpuMathNative.

Reply:
It looks for it but its not needed. Thats part of the build process itself. I will be fixing that as part of this process.